### PR TITLE
Self-Testing Clarity Docs + Fix try! bug

### PR DESCRIPTION
**Pull request by kantai**
While working on porting over examples to support moving the `u` prefix to a suffix (see, #1669), it occurred to me that it wouldn't be _too_ hard to get the docs examples to actually get tested. This PR implements that, separately from the implementation of #1669 (since that's going to be a pretty large breaking change, that should get staged before making it into our main branch), as there's definitely benefit to be had from just adding this functionality.

This PR:

1. Fixes an implementation bug discovered via the examples -- `try!` was yielding the inner error type, rather than a `(err ...)` object.
2. Adds self-testing docs, by setting up a docs-specific environment for executing the examples. If we want to integrate a sandbox into the docs-site at some point, we can set up the environment there similarly.
3. A bunch of documentation fixes discovered in this process.

*Update*

4. Adds a Github action that uses a robot account to auto-open a PR on the docs.blockstack repo if the docs output changes.

### CODE REVIEW

**Comment by kantai** [comment]
I also just added a Github action to automatically open PRs on https://github.com/blockstack/docs.blockstack/ if a push to our main branch changes the docs.

File: src/net/connection.rs [hunk 1]
```diff
@@ -1801,6 +1801,7 @@ mod test {
         pinger.join().unwrap();
     }
 
+    #[ignore] // fails intermittently when run via `cargo test`
```
Comment (jcnelson) [inline src/net/connection.rs:1804]: Can you open a separate issue for this?

Comment (kantai) [reply src/net/connection.rs:1804]: Sure -- should I remove the `#[ignore]` in this PR?

Comment (jcnelson) [reply src/net/connection.rs:1804]: No, you can leave the `#[ignore]`.  I just want to make sure we have an issue open to track this bug.

**Pull request merged**